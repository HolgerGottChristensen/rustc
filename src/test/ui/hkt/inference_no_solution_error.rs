// should fail because there should be no solution for hkt inference
// check-fail
fn test<I<%J>>(input: I<u32>, i2: I<bool>) {
}

fn main() {
    test(Some(32u32), Some("dff".to_string())); //~ ERROR cannot infer HKT parameters [E10001]
}
