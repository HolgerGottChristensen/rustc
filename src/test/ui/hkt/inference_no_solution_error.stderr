error[E10001]: cannot infer HKT parameters
  --> $DIR/inference_no_solution_error.rs:7:5
   |
LL |     test(Some(32u32), Some("dff".to_string()));
   |     ^^^^
   |
help: try annotating the function call
  --> $DIR/inference_no_solution_error.rs:7:5
   |
LL |     test(Some(32u32), Some("dff".to_string()));
   |     ^^^^

error: aborting due to previous error

